server/src/main/java/org/candlepin/dto/api/v1/PoolTranslator.java

```diff
@@ -0,0 +1,200 @@
+    @Override
+    public PoolDTO populate(Pool source, PoolDTO destination) {
+        return this.populate(null, source, destination);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PoolDTO populate(ModelTranslator modelTranslator, Pool source, PoolDTO dest) {
+        dest = super.populate(modelTranslator, source, dest);
+
+        dest.setId(source.getId());
+        dest.setType(source.getType().toString());
+        dest.setActiveSubscription(source.getActiveSubscription());
+        dest.setCreatedByShare(source.isCreatedByShare());
+        dest.setHasSharedAncestor(source.hasSharedAncestor());
+        dest.setQuantity(source.getQuantity());
+        dest.setStartDate(source.getStartDate());
+        dest.setEndDate(source.getEndDate());
+        dest.setAttributes(source.getAttributes());
+        dest.setRestrictedToUsername(source.getRestrictedToUsername());
+        dest.setContractNumber(source.getContractNumber());
+        dest.setAccountNumber(source.getAccountNumber());
+        dest.setOrderNumber(source.getOrderNumber());
+        dest.setConsumed(source.getConsumed());
+        dest.setExported(source.getExported());
+        dest.setShared(source.getShared());
+        dest.setCalculatedAttributes(source.getCalculatedAttributes());
+        dest.setUpstreamPoolId(source.getUpstreamPoolId());
+        dest.setUpstreamEntitlementId(source.getUpstreamEntitlementId());
+        dest.setUpstreamConsumerId(source.getUpstreamConsumerId());
+        dest.setProductName(source.getProductName());
+        dest.setProductId(source.getProductId());
+        dest.setProductAttributes(source.getProductAttributes());
+        dest.setStackId(source.getStackId());
+        dest.setStacked(source.isStacked());
+        dest.setDevelopmentPool(source.isDevelopmentPool());
+        dest.setMarkedForDelete(source.isMarkedForDelete());
+        dest.setDerivedProductAttributes(source.getDerivedProductAttributes());
+        dest.setDerivedProductId(source.getDerivedProductId());
+        dest.setDerivedProductName(source.getDerivedProductName());
+        dest.setSourceStackId(source.getSourceStackId());
+        dest.setSubscriptionSubKey(source.getSubscriptionSubKey());
+        dest.setSubscriptionId(source.getSubscriptionId());
+
+        // Process nested objects if we have a model translator to use to the translation...
+        if (modelTranslator != null) {
+            processNestedObjects(modelTranslator, source, dest);
+        }
+
+        return dest;
+    }
+
+    /**
+     * Utility method that translates the Pool's nested objects.
+     *
+     * @param modelTranslator the translator used to translate the nested objects.
+     *
+     * @param source the pool model object we are translating.
+     *
+     * @param dest the pool DTO object that the source model object is being translated to.
+     */
+    private void processNestedObjects(ModelTranslator modelTranslator, Pool source, PoolDTO dest) {
```

Comment:
checkstyle complained that the method was longer than 100 lines.

Reply:
Things like that are code smells to suggest maybe shortening the method. In cases where it's valid, however, it'd be preferable to simply suppress the warning rather than modifying the code to work around it in a way that doesn't actually reduce the overall footprint.

```@SuppressWarnings("checkstyle:methodlength")```